sync.go

```diff
@@ -132,7 +132,11 @@ func (w *PipeWriter) Write(p []byte) (n int, err error) {
 			}
 
 			// chunk write to fill space
-			m, err = w.b.Write(p[n : int64(n)+gap(w.b)])
```

Comment:
Also, do you mind re-basing (I only modified the travis file) so that travis will run correctly?

(not sure why my travis script broke over time)